mirdata/datasets/compmusic_otmm_makam.py

```diff
@@ -0,0 +1,249 @@
+            [self._data_home, self._track_paths["metadata"][0]]
+        )
+
+        # Get attributes
+        self.tonic = self._track_metadata.get("tonic")
+        self.makam = self._track_metadata.get("makam")
+        self.mbid = self._track_metadata.get("mbid")
+
+    @core.cached_property
+    def pitch(self):
+        return load_pitch(self.pitch_path)
+
+    @core.cached_property
+    def mb_tags(self):
+        return load_mb_tags(self.mb_tags_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            f0_data=[(self.pitch, "pitch")],
+            metadata={
+                "makam": self.makam,
+                "tonic": self.tonic,
+                "mbid": self.mbid,
+                "duration": self.mb_tags["duration"],
+                "metadata": self.mb_tags,
+            },
+        )
+
+
+@io.coerce_to_string_io
+def load_pitch(fhandle: TextIO) -> annotations.F0Data:
+    """Load pitch
+
+    Args:
+        pitch path (str): Local path where the pitch annotation is stored.
+            If `None`, returns None.
+
+    Returns:
+        F0Data: pitch annotation
+
+    """
+    time_step = 0.0029
+
+    times = []
+    freqs = []
+    reader = csv.reader(fhandle, delimiter=",")
+    for line in reader:
+        freqs.append(float(line[0]))
+
+    for i in np.arange(len(freqs)):
+        times.append(round(float(time_step * i), 4))
+
+    times = np.array(times)
+    freqs = np.array(freqs)
+    confidence = (freqs > 0).astype(float)
+    return annotations.F0Data(times, freqs, confidence)
+
+
+@io.coerce_to_string_io
+def load_mb_tags(fhandle: TextIO) -> dict:
+    """Load track metadata
+
+    Args:
+        track metadata path (str): Local path where the metadata of the track is stored.
+            If `None`, returns None.
+
+    Returns:
+        Dict: metadata of the track
+
+    """
+    mb_tags = json.load(fhandle)
+
+    return mb_tags
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The compmusic_otmm_makam dataset
+    """
+
+    def __init__(self, data_home=None):
+        super().__init__(
+            data_home,
+            index=DATA.index,
+            name="compmusic_otmm_makam",
+            track_class=Track,
+            bibtex=BIBTEX,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.cached_property
+    def _metadata(self):
+        metadata_path = os.path.join(
+            self.data_home,
+            "MTG-otmm_makam_recognition_dataset-f14c0d0",
+            "annotations.json",
+        )
+        if not os.path.exists(metadata_path):
+            raise FileNotFoundError("Metadata not found. Did you run .download()?")
+
+        metadata = {}
+        with open(metadata_path) as f:
+            meta = json.load(f)
+            for track in meta:
```

Comment:
It would be best to use a different variable name than `track` here, since it's also an attribute of `Dataset` and it could cause unexpected bugs.

Reply:
that said, I guess it won't actually cause bugs since `self.track` != `track`, but just for clarity!

Reply:
Ok! Just changed for the regular `i` so we avoid any kind of confusion.